### DISCUSSION TITLE
ngrx/component-store - select method not memoized

### DISCUSSION THREAD

**Discussion by kdmaz** [ngrx/platform#3267] [Q&A] [answered]
Hello there!

I've been using `ngrx/component-store` and I've really been enjoying it so far. I've been replacing parts of an existing app that uses `ngrx/store` with `ngrx/component-store`.

However, one issue I have run into is the `select` method is not memoized (as far as I know) like the selectors in `ngrx/store` and this has caused some performance issues during the migration. 

I was just wondering if there are any plans to memoize the `select` function in future versions of `ngrx/component-store`?

If not, what are some alternative ideas I could use to stop unnecessary emits from selectors? Is it viable to use the selectors from `ngrx/store` inside my component store instead just for the purpose of memoization?

**Comment by markostanimirovic** [comment, marked as answer]
Hey @kdmaz 

Take a look at [this comment](https://github.com/ngrx/platform/issues/3168#issuecomment-936015959).

**Reply by kdmaz** [reply]
Thank you @markostanimirovic, this solved my issue!
